CustomWeapons.cpp

```diff
@@ -247,6 +259,24 @@ HOOK_METHOD(ProjectileFactory, NumTargetsRequired, () -> int)
     return ret;
 }
 
+HOOK_METHOD(ProjectileFactory, Fire, (std::vector<Pointf> &points, int target) -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> ProjectileFactory::Fire -> Begin (CustomWeapons.cpp)\n")
+    if (blueprint->type==2 && blueprint->length==1)
```

Comment:
I'd prefer setting the grid position when aiming rather than when firing but this is acceptable, plus it also covers AI targeting. However the AI always targets the centre of the room so when the AI fires a pinpoint it's always going to snap to the same tile. I would consider adding random tile selection for the AI.

Reply:
Added random tile selection.